### ISSUE TITLE
Zero ID Trap for the unwary when creating or updating.

### ISSUE THREAD

**Issue by clydebarrow** [vincentlauvlwj/Ktorm#86]
I wrote some code like this:
````
val customer = Customers.findByEmail(email) ?: Customer()
customer.email = email
....
....
customer.address = address
if(customer.id > 0)                    // this has unexpected side-effects
     customer.flushChanges()
else
    Customers.add(customer)
````
The object was to update data if the record already existed (but not searching on the primary key) or create a new one, then using the `id` value to decide whether to insert or update. But I was finding that after an insert, the id value in the `customer` object was always zero, even though the database record had a new, auto-incremented id. It turned out that simply referencing `id ` - even though I was not assigning to it - was enough to add it to the assignment list, so the new value was not pulled from the database after insertion.
I changed the code to use a separate boolean flag to record whether I created a new record or not and that solved the problem, but the behaviour was quite unexpected.

**Comment by vincentlauvlwj** [comment]
Yes, this is a bug. I will fix it.

**Comment by vincentlauvlwj** [comment]
Fixed in version 2.7
